### DISCUSSION TITLE
plot_management: in config.yml

### DISCUSSION THREAD

**Discussion by ProblemsSender** [TownyAdvanced/Towny#5385] [Q&A] [answered]
It's confusing me at the line `block_delete` and `mayor_plotblock_delete` and `revert_on_unclaim`
It sound too familiar to me as it feels just like the same while it's separated to 3 parts. Can you help me understand the difference between them? Thanks

**Comment by TheFlagCourier** [comment, marked as answer]
Not sure what else to say that the commented configuration doesn't already address.

`block_delete` defines blocks that should have all instances deleted from a plot if the plot is deleted by normal means. This is usually when a plot is unclaimed by a Town, or through the TownyAPI.

`mayor_plotblock_delete` is similar, but only fires on the `/plot clear` command when issued by the Town's mayor. The plot remains, but the blocks listed would be replaced with `Material.AIR`. 

`revert_on_unclaim` is a nuclear option that simply rolls back plots to snapshots taken at the time they were claimed. This includes all blocks from the world's floor and ceiling - and acts similar to the Explosion Reversion function. Blocks are simply overridden with new copies of the original blocks.

All of these are overridden by world-specific configuration files. New worlds use the settings in the main config as a default.

**Reply by LlmDl** [reply]
Also further documented on the wiki: https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works#plot-regeneration--unclaimed-plots

**Reply by ProblemsSender** [reply]
May I ask more about `block_ignore` within the `revert_on_unclaim` part
1/ Is it makes the block invulnerable on revert? To me, what I thought of it is that it makes the block such as diamond_block that has been placed after the wild_plot being claimed to **keep staying there even the plot is been unclaimed**. Which means it cannot get reverted or overridden whatsoever.
2/ Is it preventing the block that is specified in `block_ignore` from generating? It's like preventing the ore to keep regenerating each time the mayor unclaimed the plot so the player can't abuse it to mine noble resources.

I was expecting it to work like that because it's too similar to think it differently, what do you think? To me, it's so cool to know if the unclaimed plot can become the ruins as it's a place you can do adventure to discover and get loot from it.

UPDATED: Can I set the timer to days, weeks, months? `speed: 1w`? Or should I set it in the first place? What would happen to the plot that gets to half the timer, does it get 50% overridden?

**Reply by LlmDl** [reply]
1/ It doesn't make it invulnerable. It means that when the plot is claimed and a snapshot is taken, any ignored block is recorded as AIR so it doesn't revert.

2/ By recording them as air, they do not regenerate a valuable block.

3/ Do not set the timer at a high amount or your server will be keeping those snapshots loaded in memory for the entire time

**Reply by ProblemsSender** [reply]
Ok, I got it, but the timer. Is there any safe value that I can use?

**Reply by LlmDl** [reply]
It should probably stay on 1 second, so that regenerations are completed in a timely fashion, but you can do what you want.

**Reply by ProblemsSender** [reply]
Will it affect my server performance if I set it high enough? I'm kinda feeling worried about `those snapshots loaded in memory for the entire time`.

**Reply by LlmDl** [reply]
Yes, this is why I'm saying not to do it.

**Reply by ProblemsSender** [reply]
hehe tysm for explaining to me all this, luv ya

**Reply by ProblemsSender** [reply]
Oh... excuse me. Today I suddenly wants to check this discussion again and I found that maybe I have misled you from my real thought about the situation.
I have edited the question above, and for convenience, I will repost it here.

May I ask more about `block_ignore` within the `revert_on_unclaim` part
1/ Is it makes the block invulnerable on revert? To me, what I thought of it is that it makes the block such as diamond_block that has been placed after the wild_plot being claimed to **keep staying there even the plot is been unclaimed**. Which means it cannot get reverted or overridden whatsoever.
2/ Is it preventing the block that is specified in `block_ignore` from generating? It's like preventing the ore to keep regenerating each time the mayor unclaimed the plot so the player can't abuse it to mine noble resources.

I was expecting it to work like that because it's **too similar to think it differently**, what do you think? To me, it's so cool to know if the unclaimed plot can **become the ruins as it's a place you can do adventure to discover and get loot from it.**

To me, `block_ignore` is like the option that you can make a block being completely ignored by the plugin, so it cannot be regenerated from the snapshot, as to be overridden by any other blocks upon revert.

This is the description in the world.txt
`# Any block Id's listed here will not be respawned. Instead, it will revert to air. This list also world on the WildRegen settings below.`
`plotManagementIgnoreIds=`
From what I read it doesn't protect the block that is specified in `plotManagementIgnoreIds` from getting overridden.

Sorry for any inconvenience.

**Reply by LlmDl** [reply]
The list if a list of blocks, that will be recorded as AIR in the plot snapshot.

If a player breaks teh diamon block after claiming the plot, it is gone forever. Even when the plot is unclaimed and everything else turns back to the pre-claimed state.

If a player does not break the diamond block, it does not need to be regenerated. When the plot is unclaimed the block stays a diamond.

**Reply by ProblemsSender** [reply]
I've tested the option
https://www.linkpicture.com/q/townyclam.png - I was build the beacon after the plot was claimed, which means no snapshot of the diamond block.
https://www.linkpicture.com/q/townyclam2.png - It was getting reverted when I unclaimed the plot, the diamond block which is not a part of the snapshot is getting overridden.

I wish the diamond block would stay there in the wild and wait for someone to loot it. UwU

**Reply by LlmDl** [reply]
The revert removes everything which wasn't there for the snapshot.

The feature turns your claimed places back to the wilderness it was before claims. So that areas can be reclaimed by someone that doesn't want to build on a ruin and only wants "fresh" land.

You want to turn off the revert on unclaim feature if you want ruins to stay.
